webpack/JobWizard/submit.js

```diff
@@ -0,0 +1,98 @@
+import { post } from 'foremanReact/redux/API';
+import { repeatTypes, JOB_INVOCATION } from './JobWizardConstants';
+import { buildHostQuery } from './steps/HostsAndInputs/buildHostQuery';
+
+export const submit = ({
+  jobTemplateID,
+  templateValues,
+  advancedValues,
+  scheduleValue,
+  selectedTargets,
+  hostsSearchQuery,
+  location,
+  organization,
+  dispatch,
+}) => {
+  const { repeatAmount, repeatType, repeatData, starts, ends } = scheduleValue;
+  const {
+    effectiveUserValue,
+    effectiveUserPassword,
+    description,
+    timeoutToKill,
+    isRandomizedOrdering,
+    timeSpan,
+    concurrencyLevel,
+    templateValues: advancedTemplateValues,
+    password,
+    keyPassphrase,
+  } = advancedValues;
+  const getCronLine = () => {
+    const [hour, minute] = repeatData.at
+      ? repeatData.at.split(':')
+      : [null, null];
+    switch (repeatType) {
+      case repeatTypes.cronline:
+        return repeatData.cronline;
+      case repeatTypes.monthly:
+        return `${minute} ${hour} ${repeatData.days} * *`;
+      case repeatTypes.weekly:
+        return `${minute} ${hour} * * ${repeatData.daysOfWeek}`;
+      case repeatTypes.daily:
+        return `${minute} ${hour} * * *`;
+      case repeatTypes.hourly:
+        return `${repeatData.minute} * * * *`;
+      case repeatTypes.noRepeat:
+      default:
+        return null;
+    }
+  };
+  const api = {
+    location,
+    organization,
+    job_invocation: {
+      job_template_id: jobTemplateID,
+      targeting_type: scheduleValue?.isTypeStatic
+        ? 'static_query'
+        : 'dynamic_query',
+      randomized_ordering: isRandomizedOrdering,
+      inputs: { ...templateValues, ...advancedTemplateValues },
+      ssh: {
+        effective_user: effectiveUserValue,
+        effective_user_password: effectiveUserPassword,
+      },
+      password,
+      key_passphrase: keyPassphrase,
+      recurrence:
+        repeatType === repeatTypes.noRepeat
+          ? null
+          : {
+              cron_line: getCronLine(),
+              max_iteration: repeatAmount,
+              end_time: ends.length ? ends.toISOString() : null,
+              purpose: '', // TODO add value after https://projects.theforeman.org/issues/33814
+            },
+      scheduling: {
+        start_at: starts.length ? starts.toISOString() : null,
```

Comment:
Would it be possible to not include the `scheduling` and `recurrence` keys in the request if the job should execute immediately? The api is confused by this and sets the job to execute in the future at time of the request, which kinda does the same thing but is less efficient than just starting it right away. An alternative would be addressing this in the api